src/Client/Systems/Events/EventsSystems.cpp

```diff
@@ -70,24 +70,30 @@ namespace Systems {
     {
         Registry &registry                                = Registry::getInstance();
         Registry::components<Types::Position> arrPosition = registry.getComponents<Types::Position>();
+        Registry::components<struct health_s> arrHealth   = registry.getComponents<struct health_s>();
         Clock &clock_                                     = registry.getClock();
         static std::size_t clockId                        = clock_.create(true);
         std::vector<std::size_t> ids =
             registry.getEntitiesByComponents({typeid(Types::Player), typeid(Types::Position)});
 
-        if (Raylib::isKeyDown(Raylib::KeyboardKey::KB_SPACE)
-            && clock_.elapsedMillisecondsSince(clockId) > waitTimeBullet) {
-            clock_.restart(clockId);
-            for (auto &id : ids) {
-                // send bullet to server
-                Nitwork::NitworkClient::getInstance().addNewBulletMsg(
-                    {static_cast<int>(arrPosition[id].x), static_cast<int>(arrPosition[id].y)},
-                    CLASSIC);
-                struct Types::Missiles missile = {
-                    .type = CLASSIC,
-                };
-                createMissile(arrPosition[id], missile);
+        if (Raylib::isKeyDown(Raylib::KeyboardKey::KB_SPACE) == false
+            || clock_.elapsedMillisecondsSince(clockId) < waitTimeBullet) {
+            return;
+        }
+
+        clock_.restart(clockId);
```

Comment:
clock_.decreaseMilliseconds(clockId, waitTimeBullet)

Reply:
this function dont even exist here, its in my pr